concurrency-limits-core/src/main/java/com/netflix/concurrency/limits/limit/window/AverageSampleWindowFactory.java

```diff
@@ -0,0 +1,36 @@
+package com.netflix.concurrency.limits.limit.window;
+
+public class AverageSampleWindowFactory implements SampleWindowFactory<ImmutableAverageSampleWindow> {
+    private AverageSampleWindowFactory() {}
+
+    public static AverageSampleWindowFactory create() {
```

Comment:
nit: Given this is a singleton we can just use a static field INSTANCE field here.